arwen/hosttest/execution_gas_test.go

```diff
@@ -0,0 +1,779 @@
+		numCallsBytes := big.NewInt(0).SetUint64(numCalls).Bytes()
+
+		test.BuildMockInstanceCallTest(t).
+			WithContracts(
+				test.CreateMockContract(test.ParentAddress).
+					WithBalance(simpleGasTestConfig.ParentBalance).
+					WithConfig(simpleGasTestConfig).
+					WithMethods(contracts.ExecOnSameCtxParentMock, contracts.ExecOnDestCtxParentMock, contracts.WasteGasParentMock),
+				test.CreateMockContract(test.ChildAddress).
+					WithBalance(simpleGasTestConfig.ChildBalance).
+					WithConfig(simpleGasTestConfig).
+					WithMethods(contracts.WasteGasChildMock),
+			).
+			WithInput(test.CreateTestContractCallInputBuilder().
+				WithRecipientAddr(test.ParentAddress).
+				WithGasProvided(simpleGasTestConfig.GasProvided).
+				WithFunction("execOnSameCtx").
+				WithArguments(test.ChildAddress, []byte("wasteGas"), numCallsBytes).
+				Build()).
+			WithSetup(func(host arwen.VMHost, world *worldmock.MockWorld) {
+				setZeroCodeCosts(host)
+			}).
+			AndAssertResults(func(world *worldmock.MockWorld, verify *test.VMOutputVerifier) {
+				verify.
+					Ok().
+					GasRemaining(expectedGasRemaining).
+					GasUsed(test.ParentAddress, simpleGasTestConfig.GasUsedByParent)
+				if numCalls > 0 {
+					verify.GasUsed(test.ChildAddress, simpleGasTestConfig.GasUsedByChild*numCalls)
+				}
+			})
+	}
+}
+
+func TestGasUsed_TwoContracts_ExecuteOnDestCtx(t *testing.T) {
+
+	for numCalls := uint64(0); numCalls < 3; numCalls++ {
+		expectedGasRemaining := simpleGasTestConfig.GasProvided - simpleGasTestConfig.GasUsedByParent - simpleGasTestConfig.GasUsedByChild*numCalls
+		numCallsBytes := big.NewInt(0).SetUint64(numCalls).Bytes()
+
+		test.BuildMockInstanceCallTest(t).
+			WithContracts(
+				test.CreateMockContract(test.ParentAddress).
+					WithBalance(simpleGasTestConfig.ParentBalance).
+					WithConfig(simpleGasTestConfig).
+					WithMethods(contracts.ExecOnSameCtxParentMock, contracts.ExecOnDestCtxParentMock, contracts.WasteGasParentMock),
+				test.CreateMockContract(test.ChildAddress).
+					WithBalance(simpleGasTestConfig.ChildBalance).
+					WithConfig(simpleGasTestConfig).
+					WithMethods(contracts.WasteGasChildMock),
+			).
+			WithInput(test.CreateTestContractCallInputBuilder().
+				WithRecipientAddr(test.ParentAddress).
+				WithGasProvided(simpleGasTestConfig.GasProvided).
+				WithFunction("execOnDestCtx").
+				WithArguments(test.ChildAddress, []byte("wasteGas"), numCallsBytes).
+				Build()).
+			WithSetup(func(host arwen.VMHost, world *worldmock.MockWorld) {
+				setZeroCodeCosts(host)
+			}).
+			AndAssertResults(func(world *worldmock.MockWorld, verify *test.VMOutputVerifier) {
+				verify.
+					Ok().
+					GasRemaining(expectedGasRemaining).
+					GasUsed(test.ParentAddress, simpleGasTestConfig.GasUsedByParent)
+				if numCalls > 0 {
+					verify.GasUsed(test.ChildAddress, simpleGasTestConfig.GasUsedByChild*numCalls)
+				}
+			})
+	}
+}
+
+func TestGasUsed_ThreeContracts_ExecuteOnDestCtx(t *testing.T) {
+
```

Comment:
Remove blank lines at the top of functions (appears in other functions as well).

Reply:
done